docs/source/reference/udfs/custom.rst

```diff
@@ -0,0 +1,142 @@
+User-Defined Functions
+======================
+
+This section provides an overview of how you can create and use a custom user-defined function (UDF) in your queries. For example, you could write a UDF that wraps around a PyTorch model.
+
+
+Part 1: Writing a Custom UDF
+------------------------------
+
+During each step, use the [UDF implementation](https://github.com/georgia-tech-db/eva/blob/master/eva/udfs/fastrcnn_object_detector.py) as a reference.
+
+1. Create a new file under `udfs/` folder and give it a descriptive name, e.g., `fastrcnn_object_detector.py`.
+
+   .. note::
+
+      UDFs packaged along with EVA are located inside the [udfs](https://github.com/georgia-tech-db/eva/tree/master/eva/udfs) folder.
+
+2. Create a Python class that inherits from `PytorchClassifierAbstractUDF`.
+
+   - The `PytorchClassifierAbstractUDF` is a parent class that defines and implements standard methods for model inference.
+   - Implement the `setup` and `forward` functions in your child class. These functions can be implemented with the help of decorators.
+
+Setup
+-----
+
+An abstract method that must be implemented in your child class. The `setup` function can be used to initialize the parameters for executing the UDF. The following parameters must be set:
+
+- `cacheable`: bool
+
+  - `True`: Cache should be enabled. The cache will be automatically invalidated when the UDF changes.
+  - `False`: Cache should not be enabled.
+
+- `udf_type`: str
+
+  - `object_detection`: UDFs for object detection.
+
+- `batchable`: bool
+
+  - `True`: Batching should be enabled.
+  - `False`: Batching is disabled.
+
+The custom setup operations for the UDF can be written inside the function in the child class. If no custom logic is required, then you can just write `pass` in the function definition.
+
+Example of the `setup` function:
+
+.. code-block:: python
+
+    @setup(cachable=True, udf_type="object_detection", batchable=True)
+    def setup(self, threshold=0.85):
+        self.threshold = threshold
+        self.model = torchvision.models.detection.fasterrcnn_resnet50_fpn(
+            weights="COCO_V1", progress=False
+        )
+        self.model.eval()
+
```

Comment:
Can we add a description of the created UDF explaining the parameters that are set? For example, here we can add,
In this example, we create a UDF for object detection and set the cacheable and batchable parameters to True. `self.threshold`, `self.model` and `self.model.eval()` are the custom logic for this UDF.